Jenkinsfile.k8s

```diff
@@ -89,41 +96,71 @@ pipeline {
             ninja -j $NP
           '''
         }
-      } 
+      }
     }
 
-    stage('Prebuild tests') {
+    stage('Run tests') {
       options {timeout(time: 90, unit: 'MINUTES')}
       steps {
         container('dealii'){
           sh '''
+            # This export avoids a warning about
+            # a discovered, but unconnected infiniband network.
             export OMPI_MCA_btl=self,tcp
+
             cd /home/dealii/build-gcc-fast/tests
-            echo "Prebuilding tests..."
-	    ninja -k 0 tests || true
-	    '''
-	}
-      }
-    }
 
-    stage('Run tests') {
-      options {timeout(time: 90, unit: 'MINUTES')}
-      steps {
-        container('dealii'){
+            # Let ninja prebuild the test libraries and run
+            # the tests to create the output files in parallel. We
+            # want this to always succeed, because it does not generate
+            # useful output (we do this further down using 'ctest', however
+            # ctest can not run ninja in parallel, so this is the
+            # most efficient way to build the tests).
+            ninja -k 0 tests || true
+          '''
           sh '''
+            # Avoid the warning described above
             export OMPI_MCA_btl=self,tcp
-            rm -f /home/dealii/build-gcc-fast/FAILED
-            cd /home/dealii/build-gcc-fast
-            ctest --output-on-failure -j4 || { touch FAILED; }
-            echo "Generating reference output..."
-            ninja generate_reference_output
-            cd $WORKSPACE
-            git checkout tests/CMakeLists.txt
+
+           cd /home/dealii/build-gcc-fast
+
+            # Output the test results using ctest. Since
+            # the tests were prebuild in the previous shell
+            # command, this will be fast although it is not
+            # running in parallel.
+	    ctest \
+              --no-compress-output \
```

Comment:
You are no longer running ctest in parallel. This is not a big issue, but will slow down this step a little bit. Is this needed?

Reply:
On the latest build of this PR, the ninja test sub-step takes 15 minutes and this ctest sub-step takes 1 minute. It's not parallel because of the issues with ctest parallelizing ninja we previously discussed.

Basically, all ctest is doing here is checking that the tests are already completed and formatting them into a Jenkins readable .xml file.